components/properties/src/script.rs

```diff
@@ -140,6 +148,58 @@ impl<'data> ScriptExtensions<'data> {
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Returns whether `script` satisfies either of: 1) matches the Script
+    /// property value; 2) is contained in the Script_Extensions property value
+    pub fn has_script(&self, code_point: u32, script: Script) -> bool {
+        if script == self.get_script_val(code_point) {
+            true
```

Comment:
This should only return true if the code_point has no Script_Extensions.
See https://github.com/unicode-org/icu/blob/main/icu4j/main/classes/core/src/com/ibm/icu/lang/UScript.java#L1225